### DISCUSSION TITLE
Homeassistant Addon: Gerätekommunikation bei einem Gerät sporadisch gestört in Raspberrymatic

### DISCUSSION THREAD

**Discussion by AzonInc** [OpenCCU/OpenCCU#2204] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Ich habe zwei Homeassistant Instanzen mit dem Raspberrymatic Addon in zwei Haushalten. Bei beiden tritt das selbe Phänomen auf, dass sporadisch bei einem Gerät (immer jeweils das selbe vom Typ HmIP-BROLL) die Kommunikation gestört ist.

Eine Instanz läuft auf einem Raspberry Pi 4 mit Funkmodul (HM-MOD-RPI-PCB) und die andere auf Generic x86 mit USB Funkmodul (RF-USB-Stick).
Am Funkmodul wird es denke ich mal nicht liegen, da die Hardware vollkommen unterschiedlich ist und es immer nur einen Rolladen Aktor betrifft.

RSSI ist bei circa -84.

Ist dieses Problem bereits bekannt? Wenn nicht, welche Informationen werden benötigt um dem ganzen auf den Grund zu gehen?

----

I have two Homeassistant instances with the Raspberrymatic addon in two households. The same phenomenon occurs with both, that sporadically with one device (always the same of type HmIP-BROLL) the communication is disturbed.

One instance runs on a Raspberry Pi 4 with wireless module and the other on Generic x86 with USB wireless module.
I don't think it's due to the radio module, since the hardware is completely different and it only affects one shutter actuator.

RSSI is around -84.

Is this problem already known? If not, what information is needed to debug that issue?

----

Raspberry Pi 4:
Rolladenaktor: HMIP-BROLL 001117099DAE69

Generic x86:
Rolladenaktor: HMIP-BROLL 0011170999CC83

### Describe the behavior you expected

Als ich das ganze noch als alleinstehende Instanzen laufen hatte, war die Gerätekommunikation nie gestört.
Seitdem es jedoch als Addon auf Homeassistant läuft, besteht dieses Problem.

### Steps to reproduce the issue

1. Homeassistant OS installieren
2. Raspberrymatic Addon hinzufügen
3. Geräte anlernen
4. Warten bis die Gerätekommunikation gestört ist

### What is the version this bug report is based on?

3.67.10.20230114

### Which base platform are you running?

intelnuc (intel NUC)

### Which HomeMatic/homematicIP radio module are you using?

HmIP-RFUSB

### Anything in the logs that might be useful for us?

```txt
Vom Generic x86 System gibt es keine Logs zu den gegebenen Uhrzeiten der Störung.

Auf dem Raspberry (Um diese Uhrzeit wurde auch die Servicemeldung mit der Gerätekommunikation ausgelöst):

Feb  6 07:45:01 de838cd8-raspberrymatic local0.warn ReGaHss: WARNING: XMLRPC 'setValue': rpcClient.isFault() failed (url: xmlrpc://127.0.0.1:32010, params: {"001117099DAE69:4","LEVEL",1.000000}, result: [faultCode:-1,faultString:"Generic error (UNREACH)"]) [CallXmlrpcMethod():iseXmlRpc.cpp:2608]
Feb  6 07:45:01 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XMLRPC 'setValue' call failed (interface: 1009, params: {"001117099DAE69:4","LEVEL",1.000000}) [CallSetValue():iseXmlRpc.cpp:1505]
Feb  6 07:45:01 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: rpc.CallSetValue failed; address = 001117099DAE69:4 [WriteValue():iseDOMdpHSS.cpp:76]
```

### Additional information

Hier sieht man auch von wann bis wann das Gerät auf dem Raspberry nicht erreichbar war (07:45 - 09:03).
![image](https://user-images.githubusercontent.com/11911587/216940117-6efa99e3-9f88-4937-82b4-fd8532ed22ec.png)

Generic x86:
![image](https://user-images.githubusercontent.com/11911587/216940840-4b795b85-ac9d-47b8-acd3-94d7a3ce4eab.png)

**Comment by jens-maus** [comment, marked as answer]
> Ist dieses Problem bereits bekannt? Wenn nicht, welche Informationen werden benötigt um dem ganzen auf den Grund zu gehen?

Funk-/Kommunikationsprobleme sind natürlich prinzipiell bekannt, liegen aber in 99,9% der Fälle an der lokalen Umgebung bzw. Gegebenheiten. Wichtig ist hierbei einen Blick auf den "Carrier Sense" zu werfen der dauerhaft < 10% liegen muss um eine stabile homematicIP Kommunikation zu ermöglichen. Ist dieser dauerhaft oder für einen gewissen zeitraum > 10% dann deutet das auf Funkstörer in der näheren Umgebung des jeweiligen Funkmodules hin. Deshalb sollte das Funkmodul auch so weit wie möglich von anderer Rechentechnik abgesetzt betrieben werden (z.B. via 2-3 m langem USB Verlängerungskabel, etc.). Auch sollte inzwischen doch hinlänglich bekannt sein, das ein RaspberryPi4 selbst ein großer Funkstörer ist und deshalb wird generell davon abgeraten das GPIO basierte Funkmodul direkt auf dem Pi4 zu betreiben und stattdessen Lösungen via USB verlängerungskabel, etc. einzusetzen.

Auch hat RaspberryMatic auf all diese Funk-basierten Probleme wenig bis keinerlei Einfluss, da die gesamte Funkkommunikation in den closed source teilen der Homematic CCU Umgebung inkl. der Gerätefirmwares stattfindet. D.h. hier kann im Rahmen des RaspberryMatic nur bedingt irgendwie geholfen werden.

Zusätzlich kann man sehr wohl davon ausgehen das diese immer wieder aufkommenden Problembeschreibungen eben rein lokaler Natur sind und nicht genereller, da es genug andere Nutzer gibt die problemlos mit den gleichen Gerätetype (wie hier der HmIP-BROLL) kommunizieren können. Insofern ist das hier kein Bug/Issue von RaspberryMatic selbst, sondern eben ein Umstand der darauf hindeutet das in der lokalen Betriebsumgebung ungünstige Verhältnisse herrschen und man stattdessen dort aktiv werden muss. Deshalb verwandle ich dieses Ticket/Issue mal besser in einen Diskussionsbeitrag da es sich strengenommen nicht um einen Bug handelt.

**Comment by AzonInc** [comment]
Danke für die schnelle Rückmeldung schonmal.

Mich wunderte es nur, dass es vor der Nutzung von der Addon Variante keine solchen  Probleme gab.
Noch skuriler war es dann, dass es wirklich nur jeweils einen Rolladen Aktor bei beiden Systemen mit vollkommen unterschiedlicher Umgebung betrifft.

Carrier Sense beim Generic System ist 0-5% und beim Raspberry n/a.
Das Funkmodul beim Generic ist über eine USB Verlängerung angeschlossen und liegt weiter vom Gerät entfernt.
Beim Raspberry wird keiner der USB Anschlüsse verwendet, um Störungen zu vermeiden. 

Während alles sonst durchgehend funktioniert und angesteuert werden kann, funktioniert immer nur der eine Rolladen Aktor nicht.
Es sind alle Möglichen Geräte im Einsatz - mehrere Rolladen Aktoren, Heizkörper Thermostate und Fenster/Türen Kontakte.

Ich werde im Detail dann nochmal nach Störenden Geräten in der Umgebung der jeweiligen Geräte schauen.
Der Strom-Schaltkasten liegt genau an der Außenwand des Raumes in dem der eine Rolladen Aktor verwendet wird (Durch diesen hätte ich eine Störung vermutet). Der Rolladenaktor unter diesem Raum an gleicher Position wiederum funktioniert ohne jede Störung.

**Reply by jens-maus** [reply]
> Mich wunderte es nur, dass es vor der Nutzung von der Addon Variante keine solchen Probleme gab.

Das kann auch einfach damit zusammenhängen das nun eben viel mehr auf dem System los ist als "nur" ein kleines RaspberryMatic. So ein HomeAssistant macht ja schon mehr Druck auf dem Kessel und damit dreht die CPU mehr auf, du hast vielleicht was am USB Port dran, nen anderen Funkstick mit dran, usw. usw. Und das kann dann eben gewisse Störfaktoren mit sich bringen.

> Noch skuriler war es dann, dass es wirklich nur jeweils einen Rolladen Aktor bei beiden Systemen mit vollkommen unterschiedlicher Umgebung betrifft.

Bis zum Beweis des Gegenteils würde ich mal behaupten das das nur Zufall ist/war.


> Carrier Sense beim Generic System ist 0-5% und beim Raspberry n/a. Das Funkmodul beim Generic ist über eine USB Verlängerung angeschlossen und liegt weiter vom Gerät entfernt. Beim Raspberry wird keiner der USB Anschlüsse verwendet, um Störungen zu vermeiden.

Das "n/a" kommt ja weil das Funkmodul das du da nutzt (HM-MOD-RPI-PCB) die Bestimmung des Carrier Sense nicht unterstützt. Dafür müsstest du ein anderen Funkmodultyp wie das HmIP-RFUSB einsetzen.
